contracts/BSCValidatorSet.sol

```diff
@@ -749,17 +749,20 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
     }
   }
 
-  function getVoteAddresses(uint length) internal view returns(bytes[] memory) {
+  function getVoteAddresses(address[] memory validators) internal view returns(bytes[] memory) {
+    // check if validatorExtraSet has been initialized
     uint n = currentValidatorSet.length;
-    bytes[] memory voteAddrs = new bytes[](length);
     if (validatorExtraSet.length != n) {
       return voteAddrs;
     }
-    uint index = 0;
-    for (uint i; i<n; ++i) {
-      if (isWorkingValidator(i)) {
-        voteAddrs[index] = validatorExtraSet[i].voteAddress;
-        index ++;
+
+    uint index;
+    uint length = validators.length;
+    bytes[] memory voteAddrs = new bytes[](length);
+    for (uint i; i<length; ++i) {
+      index = currentValidatorSetMap[validators[i]];
+      if (isWorkingValidator(index)) {
```

Comment:
There is no need to check the validator is working validator, just assign the validator's vote address is ok.

Reply:
Fixed